From b6e63f4d3502010f0affb1330a9db1d902ebecda Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 3 Mar 2016 16:00:54 +0800
Subject: [PATCH 0976/5242] MLK-12509-1 video: mipi_dsi_samsung: create a new
 dts for mipi dsi.

commit  b8b0824c551ec8eb702306b851940adf71cb6da1 from
https://source.codeaurora.org/external/imx/linux-imx.git

Create a new dts for the 'TFT3P5079E' mipi panel on
imx7d sabresd revb board.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile               |    1 +
 arch/arm/boot/dts/imx7d-sdb-mipi-dsi.dts |   30 ++++++++++++++++++++++++++++++
 2 files changed, 31 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx7d-sdb-mipi-dsi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 380c816..880607d 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -613,6 +613,7 @@ dtb-$(CONFIG_SOC_IMX7D) += \
 	imx7d-sdb-m4.dtb \
 	imx7d-sdb-sht11.dtb \
 	imx7d-sdb-qspi.dtb \
+	imx7d-sdb-mipi-dsi.dtb \
 	imx7d-sdb-reva.dtb \
 	imx7d-sdb-reva-epdc.dtb \
 	imx7d-sdb-reva-gpmi-weim.dtb \
diff --git a/arch/arm/boot/dts/imx7d-sdb-mipi-dsi.dts b/arch/arm/boot/dts/imx7d-sdb-mipi-dsi.dts
new file mode 100644
index 0000000..2ddf275
--- /dev/null
+++ b/arch/arm/boot/dts/imx7d-sdb-mipi-dsi.dts
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx7d-sdb.dts"
+
+/ {
+	mipi_dsi_reset: mipi-dsi-reset {
+		compatible = "gpio-reset";
+		reset-gpios = <&gpio1 4 GPIO_ACTIVE_LOW>;
+		reset-delay-us = <50>;
+		#reset-cells = <0>;
+	};
+};
+
+&lcdif {
+	disp-dev = "mipi_dsi_samsung";
+};
+
+&mipi_dsi {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_enet2_epdc0_en>;
+	lcd_panel = "TRULY-WVGA-TFT3P5079E";
+	resets = <&mipi_dsi_reset>;
+	status = "okay";
+};
-- 
1.7.9.5

